Last-Update: 2014-03-27
Forwarded: no
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: increase OSD's leveldb defaults
 The OSD's leveldb currently uses libleveldb's defaults for cache and
 write buffer size, which are both 4 MB.
 Increase the cache size to 128MB and the write buffer to 8MB.

  Similar change for filestore was introduced in
    https://github.com/ceph/ceph/pull/1160

--- a/src/common/config_opts.h
+++ b/src/common/config_opts.h
@@ -536,10 +536,10 @@
 OPTION(osd_op_history_duration, OPT_U32, 600) // Oldest completed op to track
 OPTION(osd_target_transaction_size, OPT_INT, 30)     // to adjust various transactions that batch smaller items
 OPTION(osd_failsafe_full_ratio, OPT_FLOAT, .97) // what % full makes an OSD "full" (failsafe)
 OPTION(osd_failsafe_nearfull_ratio, OPT_FLOAT, .90) // what % full makes an OSD near full (failsafe)
-OPTION(osd_leveldb_write_buffer_size, OPT_U64, 0) // OSD's leveldb write buffer size
-OPTION(osd_leveldb_cache_size, OPT_U64, 0) // OSD's leveldb cache size
+OPTION(osd_leveldb_write_buffer_size, OPT_U64, 8*1024*1024) // OSD's leveldb write buffer size
+OPTION(osd_leveldb_cache_size, OPT_U64, 256*1024*1024) // OSD's leveldb cache size
 OPTION(osd_leveldb_block_size, OPT_U64, 0) // OSD's leveldb block size
 OPTION(osd_leveldb_bloom_size, OPT_INT, 0) // OSD's leveldb bloom bits per entry
 OPTION(osd_leveldb_max_open_files, OPT_INT, 0) // OSD's leveldb max open files
 OPTION(osd_leveldb_compression, OPT_BOOL, true) // OSD's leveldb uses compression
